### PR TITLE
CORE-420: Add Account.validatePhrase

### PR DESCRIPTION
**Pull request by EBGToo**

### CODE REVIEW

File: Swift/BRCrypto/BRCryptoAccount.swift [hunk 1]
```diff
@@ -33,6 +33,15 @@ public final class Account {
         return Data (bytes: bytes!, count: bytesCount)
     }
 
+    internal init (core: BRCryptoAccount, uids: String, take: Bool) {
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoAccount.swift]: Can we move the UIDS down to the C layer (tracked in a separate issue/PR, so as not to gate)? In keeping with our mantra of the C API being a full fledged implementation, I don't think we want anything tracked at the wrapper layer.

Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoAccount.swift]: Yes, and see CORE-421 (the similar, big trouble-maker).

File: Swift/BRCrypto/BRCryptoAccount.swift [hunk 2]
```diff
@@ -70,27 +79,37 @@ public final class Account {
...
+    ///
+    /// Validate a phrase as a BIP-39 'paper key'; returns true if validated, false otherwise
+    ///
+    /// - Parameters:
+    ///   - phrase: the candidate paper key
+    ///   - words: A local-specific BIP-39-defined array of BIP39_WORDLIST_COUNT words.
+    ///
+    /// - Returns: true is a valid paper key; false otherwise
+    ///
+    public static func validatePhrase (_ phrase: String, words: [String]) -> Bool {
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoAccount.swift]: Phrase is a string here, but I had thought we were using Data for this (as above)?

File: Swift/BRCrypto/BRCryptoAccount.swift [hunk 3]
```diff
@@ -70,27 +79,37 @@ public final class Account {
...
+        precondition (CRYPTO_TRUE == cryptoAccountValidateWordsList (Int32(words.count)))
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoAccount.swift]: Should wordsCount be a size_t or at least an unsigned type?

File: Swift/BRCrypto/BRCryptoAccount.swift (hunk 2 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoAccount.swift]: We have another JIRA for this - CORE-399.  Not addressed yet (should be soon; I asked Ehsan about it recently and his said 'yes/okay').

File: Swift/BRCrypto/BRCryptoAccount.swift (hunk 3 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoAccount.swift]: Agreed.

File: Swift/BRCrypto/BRCryptoAccount.swift (hunk 2 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoAccount.swift]: @michael-brd For the Java Data/byte[], do you include the '\0' in the bytes?  [I think BRKeyPubKey, which uses uint8_t*, includes a '\0' in the bytes and in the returned length; in one case I saw `Data` with a '\0' - and then a string conversion produced "...abc\0"]

Comment (michael-brd) [reply Swift/BRCrypto/BRCryptoAccount.swift]: I append a \0, yes, to ensure that is null terminated.

File: Swift/BRCrypto/BRCryptoAccount.swift (hunk 1 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoAccount.swift]: CORE-432 - UIDS into BRCryptoAccount

**Pull request merged**